### PR TITLE
Wrap PMT geometry in cylindrical envelope to speed up processing

### PR DESCRIPTION
**Pull request by tgrbrooks**
tl;dr: Switched on a flag to wrap PMT in envelope, code speeds up, output looks consistent with output pre change.

I'm going to go into a fair bit of detail to preserve the process of trying to fix the worst bug I've ever had to deal with:

- Initial issue #53, generating points in center of the detector was very slow.

- Found that the slowness seemed to be a function of the distance from the PMTs, the number of PMTs in the geometry and the number of photons in the event.

- Guessed that it must be something to do with the interaction between the photons and the PMT geometry but couldn't find anything obvious in the code.

- Ran callgrind to try to find the culprit, found that a geant4 function called `SafetyForVoxelHeader` was being recursively called a lot.

Screenshot 2020-08-26 at 18 54 13

- Looked at the g4 code and it seems that at each point in the photon tracking g4 finds the distance to the nearest surface, something is wrong with the PMT geometry where it doesn't stop at the outer surface, it checks all the internal components for the PMT too.

- I noticed that in `GeoPMTFactoryBase` there was a flag called `useEnvelope`, switching this on wraps the PMT geometry logical volumes in a cylindrical physical volume which I guess acts as a consistent outer surface (this is mentioned a little here: https://github.com/rat-pac/rat-pac/issues/33).

- Switching on the flag massively reduced the simulation time but set all of the PMT IDs to zero.

- I wasn't sure if this was a fixable issue as the flag is hardcoded to false in the original rat-pac and there have been a lot of updates to the PMT geometry which aren't in the watchman rat-pac.

- I added in all of the PMT updates from the original rat-pac (mostly described here: https://github.com/rat-pac/rat-pac/pull/37) but didn't see any real change to the performance without the envelope flag.

- I reverted the PMT updates again because I found an unmerged commit (https://github.com/rat-pac/rat-pac/pull/41) which solved the problem of the PMT IDs being set to zero.

The simulation seems to give the same output (3 MeV positrons distributed evenly in detector):
charge
time
pmtid

I am still a little worried because the flag is hardcoded to false in the original rat-pac which makes me think that it might have never worked as intended and the fix I found is from a commit that isn't directly related to the envelope.

Output seems ok, but I'd feel bad if this broke a whole production so it would probably be a good idea to run it through the whole analysis pipeline to make sure it doesn't break something deep down.

### CODE REVIEW

**Comment by MarcFBergevin** [comment]
Question: Part of the official macro tries to identify the volume of interest, this volumes name as now changed. Is the envelope (inner_pmts_pmtenv_0) equivalent to the glass?

Selecting all potential PMT volumes has the following output:

```
inner_pmts_pmtenv_0, inner_pmts_body_phys, inner_pmts_inner1_phys, inner_pmts_inner2_phys, inner_pmts_dynode_phys, inner_pmts_central_gap_phys, 

inner_pmts_pmtenv_1, inner_pmts_body_phys, inner_pmts_inner1_phys, inner_pmts_inner2_phys, inner_pmts_dynode_phys, inner_pmts_central_gap_phys,

 inner_pmts_pmtenv_2, inner_pmts_body_phys, inner_pmts_inner1_phys, inner_pmts_inner2_phys, inner_pmts_dynode_phys, inner_pmts_central_gap_phys, 

inner_pmts_pmtenv_3, inner_pmts_body_phys, inner_pmts_inner1_phys, inner_pmts_inner2_phys, inner_pmts_dynode_phys, inner_pmts_central_gap_phys, 

inner_pmts_pmtenv_4, inner_pmts_body_phys, inner_pmts_inner1_phys, inner_pmts_inner2_phys, inner_pmts_dynode_phys, inner_pmts_central_gap_phys, 
```

**Comment by MarcFBergevin** [comment]
Also. I have noticed that the program sometime takes a long time to exit after all operations are completed.

**Comment by tgrbrooks** [comment]
As I understand it, the envelope volume is the same material as the mother volume, so it doesn't really exist, just acts as a continuous outer surface for g4 to calculate the distance to.

**Comment by tgrbrooks** [comment]
Has the slow exiting problem gone away? If not, is there a macro that consistently produces the issue?

**Comment by MarcFBergevin** [comment]
With this PR for water Gd-H20:
image

Gd-H20 prod_06232020 (not enough statisitics)
image

Behavior is as expected for the PMT and Liquid events.

**Review by MarcFBergevin — APPROVED** [review]
I was able to reproduce expected features in the analysis chain. 

A change to the official PMT macros and watchmakers is needed as inner_pmt_1, inner_pmt_2, ... are renamed.
